CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/Helpers/Order/OrderItemView.cs

```diff
@@ -75,51 +84,65 @@ public OrderItemView(string i_Name, OrderButtonModel i_Button, BaseLineViewModel
             {
                 if (IsInProgress)
                 {
-                    m_Labels[1].Text = "Done";
-                    m_Grid.IsVisible = false;
                     OrderStatus = 3;
-                    idNotifier.Status = 3;
-                    m_OrderStatusText = OrderStatusToString();
-                    m_Grid.Children.Remove(m_DoneButton);
-                    m_Grid.Children.Remove(m_StartButton);
-                    m_Grid.Children.Remove(m_StopButton);
+                    m_Labels[2].Text = idNotifier.StatusString;
+                    OrderStatusColor = idNotifier.StatusColor;
+                    OrderStatusString = idNotifier.StatusString;
+                    removeGridButtons();
+                    m_Grid.Children.Add(m_RestoreButton, 0, 1, 5, 6);
+                    m_OrderRestored = true;
                 }
                 else
                 {
                     Application.Current.MainPage.DisplayAlert("Order Locked", m_orderItem.m_OrderItemName, "Ok");
                 }
             });
-
-            OrderStatus = idNotifier.Status;
-            m_OrderStatusText = this.OrderStatusToString();
+            RestoreCommand = new Command(() =>
+            {
+                idNotifier.Status = 1;
+                OrderStatus = 1;
+                m_Labels[2].Text = idNotifier.StatusString;
+                OrderStatusColor = idNotifier.StatusColor;
+                OrderStatusString = idNotifier.StatusString;
+                m_Grid.Children.Remove(m_RestoreButton);
+                addGridButtons();
+            });
             m_Labels = new List<Label>();
             m_Labels.Add(new Label
```

Comment:
Everything from here up to the last line of this method **should not be here**
It should be in XAML, especially the font stuff, but mostly because these are almost static things. 
Please create them as views and give what properties are necessary such as the text for them in the background cs of them, but they should not be defined like this here.

Reply:
This has already been done but it will only come up in the next update